### PR TITLE
fix: Conditions using same variable & test but different values

### LINKED ISSUE #12: Conditions with same variable for test do not merge as list for conditions
If you follow the example in README.md for setting `allow_main` & pass in allow pr conditions. 

It'll only work for one of them. The logic for aws_iam requires all the values to be passed for each unique variable & test as a list. after the setunion I believe we need to also loop through each object in the list and merge all the values for each unique variable into one object.

To reproduce. YOu can just set the two following default values and you'll see that only one StringLike will be applied.

```hcl
module "github_oidc_repo_role" {
  source   = "../../"

  openid_connect_provider_arn = data.aws_iam_openid_connect_provider.github_oidc_provider.arn
  repo                        = "philips-labs/terraform-aws-github-oidc"
  role_name                   = "test-deploy-role"
  default_conditions          = ["allow_main", "allow_environment"]
}
```

https://github.com/philips-labs/terraform-aws-github-oidc/blob/0aeb96bd62081322ce0e39c0a8d90d29cb4a2139/main.tf#L61-L69

### PR DESCRIPTION
**Pull request by codezninja**
Fixes #12

### CODE REVIEW

**Comment by codezninja** [comment]
@ScottGuymer @npalm Any changes to get this reviewed?

File: main.tf [hunk 1]
```diff
@@ -37,7 +37,14 @@ locals {
   }] : []
 
   conditions = setunion(local.default_allow_main, local.default_allow_environment, local.default_allow_all, local.default_deny_pull_request, var.conditions)
-
+  merge_conditions = [
+    for k, v in { for c in local.conditions : "${c.test}|${c.variable}" => c... } : # group by test & variable
+    {
+      "test" : k,
+      "variable" : v[0].variable
```
Comment (npalm) [inline main.tf:44]: Looks the variable is not used

**Comment by npalm** [comment]
Will test the PR next week

File: main.tf (hunk 1 above)
Comment (codezninja) [reply main.tf]: sorry about that. I removed it and confirmed my project still worked that's using the fork

**Review by npalm — APPROVED** [review]
@codezninja thx! LGTM

**Pull request merged**